test/cache_invalidation_test.rb

```diff
@@ -46,14 +46,14 @@ def test_reload_invalidate_cached_objects
   def test_after_a_reload_the_cache_perform_as_expected
     Item.cache_has_many :associated_records, :embed => :ids
 
-    assert_equal [@baz, @bar], @record.associated_records
     assert_equal [@baz, @bar], @record.fetch_associated_records
+    assert_equal [@baz, @bar], @record.associated_records
```

Comment:
The key piece here would be `@record.reload` and then checking the result returned by `@record.fetch_associated_records` right?